### ISSUE TITLE
Move glennrp/libpng into pnggroup organization

### ISSUE THREAD

**Issue by Neustradamus** [glennrp/libpng#385]
@ctruta, @glennrp: Can you transfer https://github.com/glennrp/libpng into https://github.com/pnggroup?

Same for https://github.com/glennrp/libpng-releases.

**Comment by danoli3** [comment]
This would be up to @ctruta if possible

**Comment by Neustradamus** [comment]
@danoli3: Thanks for your reply :)

@glennrp must to do the move/transfer.

Hope that all the PNG team has rights in @pnggroup organization.

**Comment by danoli3** [comment]
Unfortunately original LibPNG author @glennrp passed away
https://sourceforge.net/p/png-mng/mailman/message/36447670/

So either this repo will need to be cloned, get GitHub support, or unsure what we can do. 
Do any members of this repo have access to add members to write access?

**Comment by Neustradamus** [comment]
@danoli3: Thanks for your comment! :)

But I think that it is only the owner who must realize this step... and it takes 2 minutes :/

I think, it is needed to contact GitHub to do the move/transfer and to explain that Glenn is dead and the new leader is @ctruta:
- https://sourceforge.net/p/png-mng/mailman/message/36447670/

When I look:
- https://sourceforge.net/p/libpng/_members/
- https://sourceforge.net/p/png-mng/_members/

I see that Greg Roelofs (@groelofs) manages png-mng on SF, I think that @ctruta can see with to have admin rights too.

Maybe good to see at the same time:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

**Comment by ctruta** [comment]
I will work with GitHub to move this repo to pnggroup/libpng, as soon as I'm done with the libpng-1.6.38 release.

**Comment by thealberto** [comment]
Hi @ctruta ,
This is an amazing news! I'd like to start to support the project so it would be cool to understand how to do it.

Thanks

**Comment by Neustradamus** [comment]
@ctruta: Nice!

Please do not forget too:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

**Comment by ctruta** [comment]
The official libpng release process used to be done via http://www.libpng.org/pub/png/libpng.html (and it still is). Moreover, the canonical location of libpng releases used to be at SourceForge (and it still is).

This was just Glenn's personal account at GitHub, rather than the PNG Group's official repo. I am still continuing to use this repo, thanks to Glenn's very graceful permission; but again, this is --still-- neither official on behalf of our group, nor is it a canonical location. It's just a personal repo in a personal account, kept up-to-date for the convenience of libpng users who are active on GitHub.

I will make the necessary changes in order to modernize whatever can be or needs to be modernized, but I'm still waiting for the GitHub officials to respond. They haven't responded yet, unfortunately.

**Comment by Neustradamus** [comment]
@ctruta: @glennrp passed away a few years ago...

I think that GitHub team will transfer these repos in the organization place:
- https://github.com/glennrp/libpng
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

**Comment by Neustradamus** [comment]
@ctruta: Have you progressed with GitHub (and Microsoft) for the transfers?

**Comment by Neustradamus** [comment]
@ctruta: Have you progressed with GitHub (and Microsoft) for the transfers?

**Comment by danoli3** [comment]
Maybe best for @ctruta to open GitHub support tickets get someone @github actioning this, mention this issue

**Comment by Neustradamus** [comment]
@ctruta: I see that there are zlib on libpng Sourceforge project, maybe good to add missing one?
- https://sourceforge.net/projects/libpng/files/zlib/

**Comment by Neustradamus** [comment]
@ctruta: I see that there is zlib on libpng Sourceforge project, maybe good to add missing one?
- https://sourceforge.net/projects/libpng/files/zlib/

**Comment by Neustradamus** [comment]
@ctruta: I see that there is zlib in libpng Sourceforge project files, maybe good to add missing one?
- https://sourceforge.net/projects/libpng/files/zlib/

**Comment by Neustradamus** [comment]
Dear @github team,

I contact you because I need a big help!

@glennrp is dead several years ago :/

The SF project and team members are here:
- https://sourceforge.net/projects/libpng/
- https://sourceforge.net/p/libpng/_members/

Cosmin Truta is here on GitHub:
- https://github.com/ctruta

It is possible to move png repositories into the pnggroup GitHub organization with current devs?
- https://github.com/pnggroup

There are:
- https://github.com/glennrp/libpng
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

Of course the current bad repo (a bad mirror) https://github.com/pnggroup/libpng must be renamed https://github.com/pnggroup/libpng-temp before removal.

Thanks a lot in advance.

Regards.

**Comment by danoli3** [comment]
This again is best handled by @ctruta the current lead of the project to handle the situation more who seems to pushing to both repos currently. 

Seems like both are set to accept updates as of December 2023. 

What are your thoughts @ctruta ? I think the community just needs clarity where to push PR's?

**Comment by Neustradamus** [comment]
@danoli3: The historical problem is that @glennrp is dead :'( and where he is, he can not done the transfers into the organization, only the owner of the repository can do it.

The solution is that a GitHub team members can do it.

**Comment by ctruta** [comment]
Hi,

When I announced libpng 1.6.40, I mentioned the new pnggroup/libpng project, and yet, few people seemed to notice.

I plan to publish libpng 1.6.41 shortly, and then I will make another announcement. I will also add various notices in various places, if that can be extra-helpful.

**Comment by Neustradamus** [comment]
@ctruta: Like you have received several emails, can you wait the GitHub answer for the transfer of repositories?

**Comment by ctruta** [comment]
> @ctruta: Like you have received several emails, can you wait the GitHub answer for the transfer of repositories?

It would be of a great convenience for me if GitHub would please acquiesce to this request.

For what is worth, I did make a formal request for repository transfer, and I was told that GitHub requires official information about the Glenn's passing. Examples include a publication of an obituary (which I do not have) or a formal note to GitHub from Glenn's family (which I would very strongly prefer to not bother them with).

**Comment by danoli3** [comment]
Glens Obituary:

> Randers-Pehrson, Glenn, who was LTC in the US ARMY, was born 30 April 1941, died 20 October 2018, and was buried in COKESBURY METHODIST, CHURCH CEMETERY, ABINGDON, MD 21009 

His attorney of estate can be reached on 1-410-569-4040 - [Terry Doyle] if GitHub want to pick up the phone and verify this.

Thanks @ctruta

**Comment by MarcusJohnson91** [comment]
He’s for sure fucking dead, stop dragging your feet @Github

https://gizmodo.com.au/2018/10/png-co-creator-glenn-randers-pehrson-passes-away-after-long-and-painful-illness/

**Comment by ctruta** [comment]
@MarcusJohnson91 not cool.
@github please moderate.

**Comment by ctruta** [comment]
To everyone reading this issue:

I will open one last ticket in one last attempt with @github to have this repository transferred to the ownership of @pnggroup. If that works, great. And if it doesn't, then I will archive this repo and I will continue with the other one that's already there.

I am also closing this in order to prevent any further abuse. I will inform everyone about the resolution of this issue by the means of PNG Group's standard communication channels.

**Comment by ctruta** [comment]
For reference, I opened GitHub Support Ticket 2527776.

**Comment by Neustradamus** [comment]
@ctruta: Like you know, mine is GitHub Support Ticket 2516229.

With repository requests to move into the https://github.com/pnggroup:
- https://github.com/glennrp/libpng
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

Maybe you can rename the current repo to libpng-temp, there will be a redirection, no lost.

**Comment by ctruta** [comment]
@Neustradamus, for now, I will continue the work on the next libpng release, business-as-usual. Should @github approve my request this time around, then, sure, I will do what they will ask me to do in order to facilitate this transfer.

**Comment by MarcusJohnson91** [comment]
Oh no I swore, #BoomerWorldProblemOn Jan 12, 2024, at 11:13 AM, Cosmin Truta ***@***.***> wrote:﻿
To everyone reading this issue:
I will open one last ticket in one last attempt with @github to have this repository transferred to the ownership of @pnggroup. If that works, great. And if it doesn't, then I will archive this repo and I will continue with the other one that's already there.
I am also closing this in order to prevent any further abuse. I will inform everyone about the resolution of this issue by the means of PNG Group's standard communication channels.

**Comment by Neustradamus** [comment]
@ctruta: Good news but there are 2 missing yet, can you send a new message to GitHub:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

It was specified in all my messages with GitHub...

Thanks in advance.

**Comment by ctruta** [comment]
> @ctruta: Good news but there are 2 missing yet, can you send a new message to GitHub:

Not now. Many other things are still TODO in our group, and they have a higher priority.

> It was specified in all my messages with GitHub...

Thank you for your interest, but this ownership transfer -- which, however important, was not at the absolute highest priority for us -- has been in the works since 2019.

The highest priority for us right at this very moment is libpng version 1.6.41, and libpng version 2 (the so-called "libpng-ng"), and (perhaps) the transfer of our group's discussion from our traditional and rather dormant mailing lists to GitHub. And, last, but absolutely not least, PNG version 3.

http://libpng.org/pub/png/pngmisc.html#lists

In conclusion:

If you want certain things to happen, I encourage you to join our PNG Group mailing lists and speak out. This is where we formulate our proposals, our agreements and disagreements, and, finally, our decisions as a whole group, ever since the inception of the PNG image format and reference library.

**Comment by Neustradamus** [comment]
@ctruta: Any progress about the transfer into this organization (like libpng repository) of:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

It is important to put the Glenn work in the good place.

**Comment by Neustradamus** [comment]
Dear @ctruta: Any progress about the transfer into this organization (like libpng repository) of:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

It is important to put the Glenn work in the good place.

**Comment by Neustradamus** [comment]
Dear @ctruta: Any progress about the transfer into this organization (like libpng repository) of:
- https://github.com/glennrp/libpng-releases
- https://github.com/glennrp/pmt

It is important to put the Glenn work in the good place.

----------------------------------

@ctruta: I have discovered that pngcheck is now in this organization :)
- https://github.com/pnggroup/pngcheck

Announcement:

[png-mng-announce] pngcheck, live at https://github.com/pnggroup/pngcheck
- https://sourceforge.net/p/png-mng/mailman/message/59131831/

```
Hello, again,

When I wrote in my previous email about being twice as happy to make software release announcements, I haven't actually counted my extra reasons to be absolutely ecstatic:

I had the honour to reconstruct a large part of the history of development of the pngcheck program, thanks to Greg Roelofs who graciously shared his offline archive containing most of the original files with original timestamps(!!)

And then, I had the honour to merge this reconstructed history (let's call it "the old history", which ended at pngcheck v3.0.3) with Chris Lilley's new development branch (let's call it "the new history", which started at pngcheck v3.0.3, not at all coincidentally).

https://github.com/pnggroup/pngcheck

As of the time of this writing, the main README file says "pngcheck unofficial of 03 Feb 2025", because it started as Chris' private fork in his own account at GitHub before having it transferred wholesale to the PNG Group's account. But make no mistake, this is as official as it gets!

--

Fun fact #1: The earliest surviving file is pngcheck version 1.2 by Alexander Lehmann, from 13th of March 1995, checking compliance against PNG draft 9. Not compatible with PNG 1.0, just so you know.
To anybody out there who stumbles upon pngcheck version 1.1 or version 1.0: please, do come forward.

Fun fact #2: Before taking over officially from Alexander Lehmann and Andreas Dilger, Greg maintained his own unofficial forks (suffixed with "grr"), which you can find in the reconstructed Git history. Point being: the present-day "unofficial-yet-official" pngcheck is not the first-ever "unofficial-yet-official" pngcheck :-)
```

cc: @svgeesus, @groelofs.

**Comment by Neustradamus** [comment]
A new ticket has been done about glennrp/pmt and glennrp/libpng-releases:
- https://github.com/pnggroup/libpng/issues/696

Note: There is too another one repository linked to: https://github.com/pnggroup/libpng/issues/694:
- https://github.com/glennrp/oss-fuzz

**Comment by ctruta** [comment]
> Note: There is too another one repository linked to: [#694](https://github.com/pnggroup/libpng/issues/694):
> 
>     * https://github.com/glennrp/oss-fuzz

That was Glenn's personal clone of Google's `oss-fuzz`, meant to be used for his own work and (perhaps) for his own submissions upstream to Google's project. I will ask GitHub to either archive it or delete it. In any case, it doesn't need to be transferred to the ownership of @pnggroup as-is. (Unless, of course, an individual user would like to clone it -- that would be absolutely fine.)

**Comment by Neustradamus** [comment]
@ctruta: I have done a reply here:
- https://github.com/pnggroup/libpng/issues/696

**Comment by jbowler** [comment]
PMT is a bit of a mess.  In the end I just copied pngcrush.c and pngcrush.h when testing it; PMT currently contains a complete copy of libpng and, IRC, zlib.  Glenn did that because he wanted to build pngcrush against specific versions of both (it won't build against current versions).

It would seem fine for someone to clone it and maintain it in its current form but I don't think it belongs in pnggroup (speaking as someone not in pnggroup); same as I don't think most of contrib/* belongs in pnggroup.
